Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -290,38 +313,35 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
 void CDMInstanceOpenCDM::updateLicense(const String& sessionId, LicenseType, const SharedBuffer& response, LicenseUpdateCallback callback)
 {
     std::string responseMessage;
-    // FIXME: This is a realy awful API, why not just take a string& ?
-    int haveMessage = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
-    GST_DEBUG("session id %s, calling callback %s message", sessionId.utf8().data(), haveMessage ? "with" : "without");
-
-    if (haveMessage) {
-        GST_TRACE("OpenCDM::Update has a message for us");
+    media::OpenCdm::KeyStatus keyStatus = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
+    GST_DEBUG("session id %s, key status is %ld", sessionId.utf8().data(), static_cast<long>(keyStatus));
+    if (keyStatus == media::OpenCdm::KeyStatus::Usable) {
+        SharedBuffer* initData = sessionIdMap.get(sessionId);
+        KeyStatusVector changedKeys;
+        GST_TRACE("OpenCDM::update returned key is usable");
+        // FIXME: Why are we passing initData here, we are supposed to be passing key IDs.
+        changedKeys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus> {*initData, openCDMKeyStatusToWebKit(keyStatus)});
```

Comment:
And note that `*initData` is the wrong here, it needs to be the key id. And we need a vector of all known keys from the CDM.

Reply:
Yes, I know, that's why I kept the FIXME, it was not my intention to fix this, just the fact that an empty message would chain up an internal error.